cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -255,6 +255,18 @@ public LocalDateTime getDateTime(String fieldName) {
     }
   }
 
+  @Nullable
+  public ByteBuffer getFixed(String fieldName) {
```

Comment:
Why need a special method? Can it be implemented as part of the normal `get` method logic?

Reply:
Since FIXED is a new logical type, I would agree that it makes sense to keep the convention we have that logical types have accessors to extract the underlying value.

Yes, i know Fixed is simply an exact-sized Byte array, but I would argue that we should keep this consistent.

Reply:
For datetime logical type, if someone calls the `get` method, it returns the timestamp as long, which logically is correct. If someone want to have the DateTime object, they call the specific `getDateTime` method.

However, for FIXED type, what's the difference between calling `getFixed` and `get`? Both would be returning a `ByteBuffer`. Does only the `getFixed` will perform a check? That seems wrong to me if only `getFixed` will check the size while `get` does not.

Reply:
For all the logicalType specific get methods are called.The method validateAndGetLogicalTypeSchema() is called inside each logicalType get method .So getFixed also has the same convention

Reply:
getFixed explicitly checks the length whereas get() does not.Also followed the convention of other getLogicalType methods

Reply:
No, they are not the same. For existing logical type, you can call `set` with any `long`, followed by `getXXX`, which will not fail (and vice versa, calling `setXXX`, followed by `get`).

For fixed, however, if you don't check length on `set`, then calling `getFixed` would fail if the length doesn't match. Hence the `set` should always check the length and we only need a `get` (without need for checking length). It is almost as if `FIXED` should be a top level type instead of logical type. BTW, why it is introduced as a logical type instead of top level type?

Reply:
@Vitalii has suggested to make it as a logical type since we are already using bytes .We are checking length in setfixed

Reply:
My original idea was not to touch schema at all as this would be pretty complex and require a lot of testing

Reply:
As discussed removed getFixed  and set Fixed methods